src/Containers/Microsoft.NET.Build.Containers/ContainerHelpers.cs

```diff
@@ -264,23 +265,45 @@ internal static bool IsValidEnvironmentVariable(string envVar)
     /// <summary>
     /// Checks if a given container image name adheres to the image name spec. If not, and recoverable, then normalizes invalid characters.
     /// </summary>
-    internal static bool NormalizeImageName(string containerImageName,
-                                         [NotNullWhen(false)] out string? normalizedImageName)
+    internal static (string? normalizedImageName, (string, object[])? normalizationWarning, (string, object[])? normalizationError) NormalizeImageName(string containerImageName)
```

Comment:
I changed this method to return warning/error information to the caller, because the wrapping we were doing in ArgumentException was being incorrectly unwrapped in the ParseContainerProperties task, leading to not-very-useful error messages.